src/main/java/org/apache/sysml/runtime/controlprogram/ProgramBlock.java

```diff
@@ -55,11 +57,14 @@
 	public static final String PRED_VAR = "__pred";
 	
 	protected static final Log LOG = LogFactory.getLog(ProgramBlock.class.getName());
-	private static final boolean CHECK_MATRIX_SPARSITY = false;
+	private static final boolean CHECK_MATRIX_SPARSITY = true;
 
 	protected Program _prog;		// pointer to Program this ProgramBlock is part of
 	protected ArrayList<Instruction> _inst;
 
+	protected SparseBlock sparseBlock = null;
+	protected SparseBlockCSR sparseBlockCSR = null;
```

Comment:
Please remove these unnecessary member variables and directly access the sparse block over the given matrix blocks.